### ISSUE TITLE
System.MissingMethodException: 'Default constructor not found for type Prism.Navigation.ErrorReportingNavigationService'

### ISSUE THREAD

**Issue by candidodmv** [dansiegel/Prism.Container.Extensions#57]
There's an error when startup the app using the "Prism.Microsoft.DependencyInjection.Forms.Extended" as can see in the StackTrace bellow:

at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00076] in <46c2fa109b574c7ea6739f9fe2350976>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache) [0x00009] in <46c2fa109b574c7ea6739f9fe2350976>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Boolean wrapExceptions, System.Threading.StackCrawlMark& stackMark) [0x00027] in <46c2fa109b574c7ea6739f9fe2350976>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) [0x0002c] in <46c2fa109b574c7ea6739f9fe2350976>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00000] in <46c2fa109b574c7ea6739f9fe2350976>:0 
  at System.Activator.CreateInstance (System.Type type) [0x00000] in <46c2fa109b574c7ea6739f9fe2350976>:0 
  at Prism.Microsoft.DependencyInjection.ConcreteAwareServiceProvider.GetConcreteImplementation (System.Type serviceType) [0x0006f] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Extensions\ConcreteAwareServiceProvider.cs:37 
  at Prism.Microsoft.DependencyInjection.ConcreteAwareServiceProvider.GetService (System.Type serviceType) [0x00000] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Extensions\ConcreteAwareServiceProvider.cs:23 
  at Prism.Microsoft.DependencyInjection.PrismContainerExtension.GetService (System.Type serviceType) [0x00000] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Extensions\PrismContainerExtension.cs:277 
  at Prism.Microsoft.DependencyInjection.NamedServiceRegistry+NamedService.GetService (System.IServiceProvider serviceProvider) [0x00000] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Extensions\NamedServiceRegistry.cs:76 
  at Prism.Microsoft.DependencyInjection.NamedServiceRegistry.GetService (System.IServiceProvider serviceProvider, System.String name, System.Type type) [0x00008] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Extensions\NamedServiceRegistry.cs:53 
  at Prism.Microsoft.DependencyInjection.PrismContainerExtension.Resolve (System.Type type, System.String name) [0x00000] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Extensions\PrismContainerExtension.cs:126 
  at Prism.Ioc.IContainerProviderExtensions.Resolve[T] (Prism.Ioc.IContainerProvider provider, System.String name) [0x00000] in d:\a\1\s\Source\Prism\Ioc\IContainerProviderExtensions.cs:24 
  at Prism.Ioc.IContainerExtensionExtensions.CreateNavigationService (Prism.Ioc.IContainerExtension containerExtension, Xamarin.Forms.Page page) [0x00000] in d:\a\1\s\Source\Xamarin\Prism.Forms\Ioc\IContainerExtensionExtensions.cs:11 
  at Prism.PrismApplicationBase.Initialize () [0x0009a] in d:\a\1\s\Source\Xamarin\Prism.Forms\PrismApplicationBase.cs:144 
  at Prism.PrismApplicationBaseExtended.Initialize () [0x00065] in d:\a\1\s\src\Prism.Forms.Extended\Platform\Droid\PrismApplication.android.cs:28 
  at Prism.PrismApplicationBase.InitializeInternal () [0x00006] in d:\a\1\s\Source\Xamarin\Prism.Forms\PrismApplicationBase.cs:84 
  at Prism.PrismApplicationBase..ctor (Prism.IPlatformInitializer platformInitializer, System.Boolean setFormsDependencyResolver) [0x00038] in d:\a\1\s\Source\Xamarin\Prism.Forms\PrismApplicationBase.cs:75 
  at Prism.PrismApplicationBase..ctor (Prism.IPlatformInitializer platformInitializer) [0x00000] in d:\a\1\s\Source\Xamarin\Prism.Forms\PrismApplicationBase.cs:57 
  at Prism.PrismApplicationBaseExtended..ctor (Prism.IPlatformInitializer platformInitializer) [0x00000] in d:\a\1\s\src\Prism.Forms.Extended\PrismApplicationBaseExtended.cs:29 
  at Prism.Microsoft.DependencyInjection.PrismApplication..ctor (Prism.IPlatformInitializer platformInitializer) [0x00000] in d:\a\1\s\src\Prism.Microsoft.DependencyInjection.Forms.Extended\PrismApplication.cs:14 
  at App.Core.App..ctor (Prism.IPlatformInitializer initializer) [0x00007] in C:\App.Core\App.xaml.cs:100 
  at App.Droid.Activities.MainActivity.OnCreate (Android.OS.Bundle savedInstanceState) [0x000a4] in C:\VS Projects\App.Droid\Activities\MainActivity.cs:76 
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00011] in <11a340ccc8de43f09c97400139266ef5>:0 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.12(intptr,intptr,intptr)

**Comment by dansiegel** [comment]
@candidodmv can you provide a repro app please

**Comment by candidodmv** [comment]
Hi @dansiegel , unfortunately there's sensible information in that project that turn impossible to be shared, but if I can gave you more details about the error, I'm available

**Comment by dansiegel** [comment]
@candidodmv you don't need to share that project...just create a sample that reproduces the issue

**Comment by candidodmv** [comment]
All right @dansiegel I could simulate the error in the following project:

https://github.com/candidodmv/AcrDialogsPrismModule/tree/reproduceMicrosoftIoCEror

**Comment by dansiegel** [comment]
I suppose depending on how you look at it... good news/bad news... I was able to duplicate the issue and am looking at fixes for the next release. Thanks for bringing it to my attention @candidodmv

**Comment by dansiegel** [comment]
I'm providing it because I know people would like it. That said I personally have no intention of using it and will continue to use DryIoc which I consider to be a far better solution. 

Also keep in mind that the only reason that this works is that I've hacked at the ServiceProvider to compensate for the fact that it cannot resolve Concrete types that weren't registered (which no ViewModel ever unless you've explicitly registered it. Also I've had to provide a hack to compensate for the fact that it cannot handle named services by itself and that we do actually expect an exception to be thrown for Prism but Shiny expects the IServiceProvider to just return null if nothing is available.
